Hosted CE doc (SOFTWARE-3445) (#507)
opensciencegrid/docs: docs/compute-element/hosted-ce.md

Applying for an OSG Hosted CE
-----------------------------

Before making any system changes, you should do the following steps:

-   Fill out the [cluster integration questionnaire](http://goo.gl/forms/8OukxsyG6KBSGHuR2)
    so that the OSG team has basic information about your cluster

-   Email [help@opensciencegrid.org](mailto:help@opensciencegrid.org)
    to set up a consultation call with the OSG team,
    in order to discuss how you would like to contribute to the OSG;
    for example, the number of OSG jobs that should run, what resource limits you have,
    or which science communities you support

After the consultation, do the following:

-   Create a Unix account on the submit server for use by the OSG CE

-   Install public SSH keys for the account

Once this is done, the OSG team will:

-   Configure the OSG CE with your system details

-   Validate operation with a set of test jobs

-   Configure central OSG services to schedule jobs

Optionally, we can assist you in installing and setting up the Squid and OASIS software on your cluster
in order to support application software repositories.
This will allow a broader set of jobs to run on your cluster.

> Technical Setup

Reviewer: ```suggestion
Providing the OSG Worker Node client (HTCondor batch systems only)
```

---------------

> An OSG Hosted CE site needs to have access to the worker node client.

Reviewer: All OSG sites need to provide the worker node client

Author: (updated)

On a Hosted CE with a non-HTCondor batch system, this is often done by unpacking the worker node client tarball into the

> users home directories, if this is on a shared filesystem.

Reviewer: I think we want to hide some of the gory details from the user and say something along the lines that this is normally handled by OSG staff in the hosted CE case but it requires home directories shared across the cluster.

But for sites with an HTCondor batch system, often there is no shared filesystem set up.
In that case, the site can either set up a shared filesystem, or install the OSG worker node client.

> This can be obtained from the osg-wn-client rpm, the osg-wn tarball, or used directly from OASIS.

Reviewer: Even though it's the next section, turn OASIS into a link to it

Reviewer: We should also say something along the lines of: if it is easier for your site to install and maintain the WN client than to set up a shared file system, you have the following options:

-   Create a Unix account on the submit server for use by the OSG CE

-   Install public SSH keys for the account

Once this is done, the OSG team will:

-   Configure the OSG CE with your system details

-   Validate operation with a set of test jobs

-   Configure central OSG services to schedule jobs

Optionally, we can assist you in installing and setting up the Squid and OASIS software on your cluster
in order to support application software repositories.
This will allow a broader set of jobs to run on your cluster.

Technical Setup
---------------

An OSG Hosted CE site needs to have access to the worker node client.
On a Hosted CE with a non-HTCondor batch system, this is often done by unpacking the worker node client tarball into the
users home directories, if this is on a shared filesystem.
But for sites with an HTCondor batch system, often there is no shared filesystem set up.
In that case, the site can either set up a shared filesystem, or
[install the OSG worker node client](/worker-node/install-wn).
This can be obtained from the [osg-wn-client rpm](/worker-node/install-wn),
the [osg-wn tarball](/worker-node/install-wn-tarball),

> or installed directly from [OASIS](/worker-node/install-wn-oasis).

Reviewer: I think I'd prefer this in a bulleted list

OSG Hosted CE
=============

Introduction
------------

An OSG Compute Element (CE) is the entry point for jobs coming from the OSG;
it handles authorization and delegation of jobs to your existing campus HPC/HTC cluster.

Many sites set up their compute element locally.
As an alternative, OSG offers a hosted CE option
wherein the OSG team will host and operate the HTCondor compute element,
and configure it for the science communities that you choose to support.

> 

Reviewer: Let's add a sentence here saying that the Hosted-CE can support thousands of concurrent job submissions

Author: updated

> This document explains the requirements and the procuedure for obtaining a hosted CE.

Reviewer: Could you run a spell-checker?

```suggestion
This document explains the requirements and the procedure for obtaining a hosted CE.
```

If you wish to have a local compute element instead,

> see page for [installing HTCondor-CE](/compute-element/install-htcondor-ce).

Reviewer: Let's change this to something like:

> If you wish to run your own local compute element or expect to support more than 10,000 concurrently running OSG jobs, see this page for...

Author: updated

Author: updated

![managed services diagram](/compute-element/img/managed_services_diagram.png)

Before Starting
---------------

Before preparing your cluster for OSG jobs, consider the following requirements:

-   An existing compute cluster with a [supported batch system](/index.md#prepare-the-batch-system)
    running on a [supported operating system](/release/supported_platforms)

-   Outbound network connectivity from the compute nodes (they can be behind NAT)

-   A Unix account on your cluster's submit server, accessible via an SSH key
    The OSG CE will use this account to automatically submit jobs,
    so it must also have permissions to submit jobs to the batch system

-   If your batch system is not HTCondor,

>     there must be a shared file system between the submit server and the compute nodes

Reviewer: Let's add a `See this section for details` with a link to the section you added below

Author: updated

Author: updated

Which Science Communities and Institutions am I Supporting?
-----------------------------------------------------------

The OSG provides monitoring to view which communities are accessing your site, their fields of science, and home institution.
Below is an example of the monitoring views that will be available for your cluster.

![monitoring graphs](/compute-element/img/monitoring_graphs.png)

Security
--------

OSG takes multiple precautions to maintain security and prevent unauthorized
usage of resources:

-   Access to the OSG system with SSH keys are restricted to the OSG staff maintaining them
-   Users are carefully vetted before they are allowed to submit jobs to OSG
-   Jobs running through OSG can be traced back to the user that submitted them
-   Job submission can quickly be disabled if needed
-   OSG staff are readily contactable in case of an emergency,
    through email at <mailto:help@opensciencegrid.org>

Applying for an OSG Hosted CE
-----------------------------

Before making any system changes, you should do the following steps:

> -   Fill out the [cluster integration questionnaire](http://goo.gl/forms/8OukxsyG6KBSGHuR2)

Reviewer: Updated link (unfortunately the short-form seems to expire after some time)

```suggestion
-   Fill out the [cluster integration questionnaire](https://docs.google.com/forms/d/e/1FAIpQLSf_pJ3Xlf2N9v5I1GK9oA9HVIxNeRikEq_T1GY0ijaUMy9cIA/viewform)
```

    so that the OSG team has basic information about your cluster

-   Email [help@opensciencegrid.org](mailto:help@opensciencegrid.org)
    to set up a consultation call with the OSG team,
    in order to discuss how you would like to contribute to the OSG;
    for example, the number of OSG jobs that should run, what resource limits you have,
    or which science communities you support

After the consultation, do the following:

> -   Create a Unix account on the submit server for use by the OSG CE

Reviewer: ```suggestion
-   Create a Unix account for OSG jobs on the submit server and across the cluster as necessary
```

> -   Install public SSH keys for the account

Reviewer: ```suggestion
-   Install OSG-provided public SSH keys for the account
```

Once this is done, the OSG team will:

> -   Configure the OSG CE with your system details

Reviewer: Let's replace OSG CE with "Hosted CE" everywhere in the doc and make sure Hosted CE is capitalized consistently

Author: Updated

-   Validate operation with a set of test jobs

-   Configure central OSG services to schedule jobs

Optionally, we can assist you in installing and setting up the Squid and OASIS software on your cluster
in order to support application software repositories.

> This will allow a broader set of jobs to run on your cluster.

Reviewer: Link to the section below about OASIS somewhere in this paragraph

Author: updated

Providing the OSG Worker Node client (HTCondor batch systems only)
------------------------------------------------------------------

> All OSG sites need to provide the worker node client.

Reviewer: ```suggestion
All OSG sites need to provide the OSG Worker Node Client on each worker node in the cluster.
```

This is normally handled by OSG staff for a Hosted CE, but requires shared home directories across the cluster.
However, for sites with an HTCondor batch system, often there is no shared filesystem set up.

> If it is easier for your site to install and maintain the worker node client than to set up a shared file system, you have the following options:

Reviewer: ```suggestion
If it is easier for your site to install and maintain the Worker Node Client on each worker node than to set up a shared file system, you have the following options:
```

> -   Install the [osg-wn-client rpm](/worker-node/install-wn)

Reviewer: ```suggestion
-   Install the [Worker Node Client from RPM](/worker-node/install-wn)
```

> -   Install the [osg-wn tarball](/worker-node/install-wn-tarball)

Reviewer: ```suggestion
-   Install the [Worker Node Client from tarball](/worker-node/install-wn-tarball)
```

> -   Install the worker node client from [OASIS](/worker-node/install-wn-oasis)

Reviewer: ```suggestion
-   Install the Worker Node Client from [OASIS](/worker-node/install-wn-oasis)
```
